### ISSUE TITLE
OCRift - Plugin breaks if you end/start a game at huge essense portal

### ISSUE THREAD

**Issue by AdamSaketume26** [y-fletch/ocplugins#2]
- Take portal to huge remains
- Game ends
- Bot is stuck there with the action to take weak cell
- Game starts and then it switches to mine the huge essence
- Once it leaves through the portal, it stops with the debug action of nothing

I turned plugin off, finished the run, and started it back on. It worked well but once it crafts the essence blocks and it's time to craft it breaks again.

**Comment by y-fletch** [comment]
Yeah I've noticed this also happens when inside one of the altars when the guardian dies. I think it could be fixed by listening for the "guardian defeated" message to reset the game time, and adding extra conditions to the exit portal/exit altar actions to always leave if pre-game (and adjusting priority)

**Comment by AdamSaketume26** [comment]
> adding extra conditions to the exit portal

maybe dont leave if the game ends and you are at the huge essence. Instead I suggest waiting there till the next game start, drop the extra charged cells, mine the huge essence, leave when full and make enough space to pick up a weak cell and 10 uncharged ones. Continue as normal.
